dhctl/pkg/kubernetes/actions/deckhouse/delete.go

```diff
@@ -49,6 +72,33 @@ func DeleteDeckhouseDeployment(kubeCl *client.KubernetesClient) error {
 	})
 }
 
+func DeleteDeckhouseStorageCRs(kubeCl *client.KubernetesClient) error {
+	return retry.NewLoop("Delete Deckhouse Storage CRs", 45, 5*time.Second).WithShowError(false).Run(func() error {
+		for _, cr := range d8storageConfig {
+			ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+			defer cancel()
```

Comment:
Do not defer in a loop, it is a resource leak, and it will execute at the end of the function, not at the end of each interation.

Reply:
Well, I can make explicit cancel() of context where needed. I think to use timeout of 30 seconds for deletion of each resource type

Reply:
Just move the loop body into a separate func and defer there. 30 secs seems fine for k8s apiserver.

Reply:
ok, I've split it into three functions